flink-runtime-web/src/main/java/org/apache/flink/runtime/webmonitor/history/HistoryServerArchiveFetcher.java

```diff
@@ -231,18 +243,20 @@ public void run() {
 	 * <p>For the display in the HistoryServer WebFrontend we have to combine these overviews.
 	 */
 	private static void updateJobOverview(File webDir) {
-		File webOverviewDir = new File(webDir, "overviews");
-		try (JsonGenerator gen = jacksonFactory.createGenerator(HistoryServer.createOrGetFile(webDir, "joboverview"))) {
+		File webOverviewDir = new File(webDir, "/jobs/");
+		try (JsonGenerator gen = jacksonFactory.createGenerator(HistoryServer.createOrGetFile(webDir, "/jobs/overview"))) {
 			gen.writeStartObject();
-			gen.writeArrayFieldStart("running");
-			gen.writeEndArray();
-			gen.writeArrayFieldStart("finished");
+			gen.writeArrayFieldStart("jobs");
 
 			File[] overviews = new File(webOverviewDir.getPath()).listFiles();
 			if (overviews != null) {
 				for (File overview : overviews) {
-					JsonNode root = mapper.readTree(overview);
-					JsonNode finished = root.get("finished");
+					if (overview.isFile()) {
+						continue;
+					}
+					File file = new File(overview.getAbsolutePath() + "/overview.json");
+					JsonNode root = mapper.readTree(file);
```

Comment:
A much safer implementation would be to parse these into `MultipleJobsDetails`, merge them and write them back out.